### ISSUE TITLE
Weird clipboard behavior on remote hosts running version 2.9

### ISSUE THREAD

**Issue by hisnawi** [tmux/tmux#2316]
* tmux version (`tmux -V`).
2.9a
* Platform (`uname -sp`).
Darwin i386
* $TERM inside and outside of tmux (`echo $TERM`).
screen-256color

Description:

I use tmux over ssh on MacOS remote systems to run regressions. Some of these systems have tmux 2.4, 2.9a or 3.1b.
I noticed when trying to copy text from remote to local emacs only for the hosts running 2.9a version I can only copy around 1000 lines of text. If I go over that limit then the "previous" copy would be pasted.
When upgrading the same tmux on the host which was running 2.9a to 3.1b I don't see this behavior and I can copy as much text as I want with no limit. Same applies to version 2.4, no problem there.

The problem is these are hosts used by many others and I can't upgrade their system files (not advisable). So I rather figure the issue with tmux version 2.9a and adapt a different setup? or workaround a possible bug.

This is the related copy/paste code in the remote tmux config file:

```
# ================================================
# ===     Copy mode, scroll and clipboard      ===
# ================================================
set -g @copy_use_osc52_fallback on

# Prefer vi style key table
setw -g mode-keys vi
bind-key -T copy-mode-vi 'v' send -X begin-selection
# Could also use 'send -X copy-pipe-and-cancel'
bind-key -T copy-mode-vi 'y' send -X copy-pipe 'pbcopy -in -selection clipboard'
bind p paste-buffer
# buffer
bind Space choose-buffer

# When scrolling with mouse wheel, reduce number of scrolled rows per tick to "2" (default is 5)
bind -T copy-mode-vi WheelUpPane       select-pane \; send-keys -X -N 2 scroll-up
bind -T copy-mode-vi WheelDownPane     select-pane \; send-keys -X -N 2 scroll-down

# Mouse drag acts normally
bind -T copy-mode-vi MouseDragEnd1Pane send -X copy-selection

```

**Comment by nicm** [comment]
Not much changed with OSC 52 between 2.9a and 3.1b so possibly this is outside tmux (libevent version, kernel version). Are you sure it works with 3.1b on exactly the same box as it didn't work with 2.9a?

In any case, 2.9a is no longer supported, so you will need to reproduce with 3.1b.

**Comment by hisnawi** [comment]
Yes, I am very sure it works on 3.1b on exactly same box.
I understand 2.9a is no longer supported. I was hoping for pointers I could look at to help me solve this issue.

**Comment by nicm** [comment]
If you can find which commit fixed it between 2.9a and 3.1b then you might be able to figure out a workaround but it is probably unlikely.
